cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -45,18 +45,18 @@ var (
 
 // SpanHandlerBuilder builds span (Jaeger and zipkin) handlers
 type SpanHandlerBuilder interface {
-	BuildHandlers() (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error)
+	BuildHandlers(cOps *CollectorOptions) (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error)
 }
 
 // NewSpanHandlerBuilder returns a span handler
-func NewSpanHandlerBuilder(opts ...basicB.Option) (SpanHandlerBuilder, error) {
+func NewSpanHandlerBuilder(flagOpts *flags.Flags, opts ...basicB.Option) (SpanHandlerBuilder, error) {
```

Comment:
Is it possible to separate the concerns? I.e. only main() needs to know about flags/viper, and here we just take `*CollectorOptions` in the constructor (not in the interface above)?